test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpCodeFixVerifier`2.cs

```diff
@@ -32,6 +33,7 @@ public Test ()
 					compilationOptions = compilationOptions!.WithSpecificDiagnosticOptions (
 						compilationOptions.SpecificDiagnosticOptions.SetItems (CSharpVerifierHelper.NullableWarnings));
 					solution = solution.WithProjectCompilationOptions (projectId, compilationOptions);
+					solution = solution.WithProjectMetadataReferences (projectId, new List<MetadataReference> { MetadataReference.CreateFromFile (typeof (object).Assembly.Location) });
```

Comment:
I don't understand why this is necessary...

Reply:
Well, it swaps the metadataReferences in the project from netcoreapp3.1 to whatever the linker tests are running in this case netcoreapp6.0. If we use the available [ReferenceAssemblies.Net.Net50](https://github.com/dotnet/roslyn-sdk/blob/main/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs#L868) the RequiresAssemblyFiles attribute would be missing since it was introduced in 6.0 and there is no ReferenceAssemblies.Net.Net60

Reply:
Not sure if the methodology is to use ReferenceAssemblies.Net.Net50 and add the additional references manually until net6.0 gets published the ReferenceAssemblies list gets updated and then we can update everything. For me, it seemed like running against the tfm of the tests is not a bad option.

Reply:
Ah, I see, you're adding another corelib as a reference (System.Private.CoreLib). That's pretty messy. Let's see if we can do this properly. In `VerifyCodeFixAsync` we create a new `Test` -- I think we can set `ReferenceAssemblies` to `ReferenceAssemblies.Net50`. We'll still have to define the `RequiresAssemblyFiles` attribute ourselves, but that doesn't seem to be a big deal.

Reply:
Ok I will change both the analyzer and codefix references assemblies to use ReferenceAssemblies.Net.Net50, which would mean we revert https://github.com/mono/linker/pull/1922

Reply:
If we still want to test that specific APIs produce correct warnings (e.g., `Assembly.Location`) we have to keep the md reference, no? At least until we annotate everything in the runtime and we can start using `ReferenceAssemblies.Net60`.